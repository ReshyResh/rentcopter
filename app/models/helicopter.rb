class Helicopter < ApplicationRecord
  mount_uploader :avatar, AvatarUploader
  belongs_to :user
  has_many :reservations, dependent: :destroy

  validates :name, :description, :model, :image, presence: true
end
