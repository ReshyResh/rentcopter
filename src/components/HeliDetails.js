import { PropTypes } from 'prop-types';

const HeliDetails = (props) => {
  const {
    helicopter: {
      speed,
      capacity,
      autonomy,
      price_per_day: price,
    },
  } = props;

  return (
    <ul className="heli-details-list">
      <li>
        <span>Speed:</span>
        <span>{`${speed} KTS`}</span>
      </li>
      <li>
        <span>Capacity:</span>
        <span>{`${capacity} People`}</span>
      </li>
      <li>
        <span>Autonomy:</span>
        <span>{`${autonomy} NM`}</span>
      </li>
      <li>
        <span>Price:</span>
        <span>{`${price} USD`}</span>
      </li>
    </ul>
  );
};

HeliDetails.propTypes = {
  helicopter: PropTypes.instanceOf(Object).isRequired,
};

export default HeliDetails;
