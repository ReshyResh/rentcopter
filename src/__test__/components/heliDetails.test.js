import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import HeliDetails from '../../components/HeliDetails';
import helicopter from '../mockHeli';
import store from '../../redux/configureStore';

it('Helicopter details component renders correctly', () => {
  const tree = renderer
    .create(<Router>
      <Provider store={store}>
        <HeliDetails helicopter={helicopter} />
      </Provider>
            </Router>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
