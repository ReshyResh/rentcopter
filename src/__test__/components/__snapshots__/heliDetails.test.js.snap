// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helicopter details component renders correctly 1`] = `
<ul
  className="heli-details-list"
>
  <li>
    <span>
      Speed:
    </span>
    <span>
      260 KTS
    </span>
  </li>
  <li>
    <span>
      Capacity:
    </span>
    <span>
      6 People
    </span>
  </li>
  <li>
    <span>
      Autonomy:
    </span>
    <span>
      350 NM
    </span>
  </li>
  <li>
    <span>
      Price:
    </span>
    <span>
      500 USD
    </span>
  </li>
</ul>
`;
